src/components/network/network-table.js

```diff
@@ -514,53 +563,51 @@ const NetworkTable = (props) => {
         [rowFilter]
     );
 
-    const handleOpenPopupSelectColumnNames = () => {
+    const handleOpenPopupSelectColumnNames = useCallback(() => {
+        setBackupSelectedColumnsNames(selectedColumnsNames);
         setPopupSelectColumnNames(true);
-    };
+    }, [selectedColumnsNames]);
 
-    const handleClosePopupSelectColumnNames = () => {
+    const handleCancelPopupSelectColumnNames = useCallback(() => {
+        setSelectedColumnsNames(backupSelectedColumnsNames);
+        setBackupSelectedColumnsNames(null);
         setPopupSelectColumnNames(false);
-    };
-
-    const handleSaveSelectedColumnNames = () => {
+    }, [backupSelectedColumnsNames]);
+
+    const handleSaveSelectedColumnNames = useCallback(() => {
+        // Optimization to reduce database access
+        if (selectedColumnsNames !== backupSelectedColumnsNames) {
+            updateConfigParameter(
+                COLUMNS_PARAMETER_PREFIX_IN_DATABASE + TABLES_NAMES[tabIndex],
+                JSON.stringify([...selectedColumnsNames])
+            );
+        }
+        setBackupSelectedColumnsNames(null);
         setPopupSelectColumnNames(false);
-    };
+    }, [tabIndex, selectedColumnsNames, backupSelectedColumnsNames]);
 
     const handleToggle = (value) => () => {
-        const newChecked = new Set(checkedColumnsNames[tabIndex]);
-        if (checkedColumnsNames[tabIndex].has(value)) {
+        const newChecked = new Set([...selectedColumnsNames]); // copy
```

Comment:
why do you need a copy to create a new set anyway ?

Reply:
Because you can't modify directly the state variable `selectedColumnsNames` (immutable)
Is it true ?

Reply:
new Set(selectedColumnsNames.values())